xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp

```diff
@@ -139,18 +139,19 @@ bool CRendererDXVA::Configure(const VideoPicture& picture, float fps, unsigned o
 
       if (m_processor->IsFormatConversionSupported(dxgi_format, dest_format, picture))
       {
-        const auto settings = CServiceBroker::GetSettingsComponent()->GetSettings();
+        if (DXVA::CProcessorHD::IsSuperResolutionSuitable(picture))
```

Comment:
Please correct it in the way it was discussed in https://github.com/xbmc/xbmc/pull/23336#discussion_r1221208631


Move `SupportsVideoSuperResolution()` first condition

```c++
        if (DX::Windowing()->SupportsVideoSuperResolution())
        {
          const auto settings = CServiceBroker::GetSettingsComponent()->GetSettings();
          if (!settings)
            return true;

          if (settings->GetBool(CSettings::SETTING_VIDEOPLAYER_USESUPERRESOLUTION) &&
              CProcessorHD::IsSuperResolutionSuitable(picture))
          {
            m_processor->TryEnableVideoSuperResolution();
          }
        }
```

also namespace `DXVA::` can be removed now

Reply:
makes more sense yes.